SirMorland/huld-hub
# Feat/GitHub docker setup

## Description
Add end-to-end workflow which is intended for end-to-end testing (robot framework).
The workflow spins up backend and frontend docker containers, waits them to start and then closes them.

As a side effect I created a ping method to backend which simply responds "pong", and ping-page to frontend which calls this endpoint and shows the result on the page.
These are to later test that frontend is able to communicate with backend.

## Considerations and implementation
I wrapped the `App` component of frontend inside React `BrowserRouter`. `App` should now have props provided by `BrowserRouter` (`match`, `location` and `history`).

### How to test
To test end-to-end workflow, simply take a look if the workflows pass.
To test ping-pong functionality, turn on backend and frontend and go to [http://localhost:3000/ping](http://localhost:3000/ping). You should see a page with text "Waiting for response..." (for very briefly) and then "pong". If your frontend can't connect to your backend, the page could read somethin like "Not found".

### Test(s) added 
Test that GitHub can spin up the docker containers and wait for them to start.
